src/modules/dashboard/components/CoinMachine/CoinMachine.tsx

```diff
@@ -269,6 +316,10 @@ const CoinMachine = ({
                 colony={colony}
                 isCurrentlyOnSale={hasSaleStarted}
                 isSoldOut={isSoldOut}
+                salePriceData={salePriceData}
+                maxUserPurchaseData={maxUserPurchaseData}
+                loadingSalePrice={loadingSalePrice}
+                loadingMaxUserPurchase={loadingMaxUserPurchase}
```

Comment:
we had this conversation already. hooks was introduced to avoid props drilling. If you use same query inside BuyTokens (instead of passing it as a props) its cost free so no point of moving it outside.

Reply:
I don't know what you are referring to. I didn't touch the CoinMachine file in my branch. Maybe you reviewed it when it was not rebased. Now it's rebased, you can check again.